File: LEET-CODE/7.REVERSE_INTEGER/REVERSE_INTEGER.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {                      
public:
    int reverse(int x) {
        int r=0;      // decleare r 
        try {
            while (x) {
                r = r * 10 + x % 10;
                x = x / 10;
            }
        } catch (const overflow_error& e) {
            return 0;
        } // check range if r is outside the range then return 0  
        return int(r);  // if r in the 32 bit range then return r
    }
}; 
//Not proper solution

File: LEET-CODE/7.REVERSE_INTEGER/REVERSE_INTEGER.py
def reverse(x: int):
    # Check if the number is negative
    if x < 0:
        x = -x  # Convert the number to its positive form
        x = int(str(x)[::-1])  # Reverse the digits of the positive number
        x = -x  # Convert the number back to negative
    else:
        x = int(str(x)[::-1])  # Reverse the digits of the positive number

    # Check if the reversed number is within the 32-bit signed integer range
    if x < -2 ** 31 or x > 2 ** 31 - 1:
        return 0

    return x
#Not proper solution